spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -371,6 +383,9 @@ public boolean tryLock(long time, TimeUnit unit) throws InterruptedException {
 				}
 				if (!acquired) {
 					localLock.unlock();
+					if (this.reLock <= 0) {
+						removeThreadLocal();
+					}
```

Comment:
Yeah, but that's for all tries for get lock, not only `tryLock`, and btw my solution will not work :(
```
Lock a1 = obtain('a'); 
a1.tryLock(); //false
Lock a2 = obtain('a');
```
a1 should be equal to a2 but it will not with my 'fix', so need smth like soft reference